skynet/dgpu/network.py

```diff
@@ -13,40 +13,72 @@
 import anyio
 import httpx
 
-from PIL import Image, UnidentifiedImageError
+from PIL import (
+    Image,
+    # UnidentifiedImageError,  # TODO, remove?
+)
 
 from leap.cleos import CLEOS
 from leap.protocol import Asset
-from skynet.constants import DEFAULT_IPFS_DOMAIN, GPU_CONTRACT_ABI
-
-from skynet.ipfs import AsyncIPFSHTTP, get_ipfs_file
-from skynet.dgpu.errors import DGPUComputeError
-
-
-REQUEST_UPDATE_TIME = 3
-
-
-async def failable(fn: partial, ret_fail=None):
+from skynet.constants import (
+    DEFAULT_IPFS_DOMAIN,
+    GPU_CONTRACT_ABI,
+)
+
+from skynet.ipfs import (
+    AsyncIPFSHTTP,
+    get_ipfs_file,
+)
+# TODO, remove?
+# from skynet.dgpu.errors import DGPUComputeError
+
+
+REQUEST_UPDATE_TIME: int = 3
+
+
+# TODO, consider using the `outcome` lib instead?
+# - it's already purpose built for exactly this, boxing (async)
+#  function invocations..
+# |_ https://outcome.readthedocs.io/en/latest/api.html#outcome.capture
+async def failable(
+    fn: partial,
+    ret_fail=None,
+):
     try:
         return await fn()
-
     except (
         OSError,
         json.JSONDecodeError,
         anyio.BrokenResourceError,
         httpx.ReadError,
         leap.errors.TransactionPushError
-    ) as e:
+    ):
         return ret_fail
 
 
+# TODO, again the prefix XD
+# -[ ] better name then `GPUConnector` ??
+# |_ `Compute[Net]IO[Mngr]`
 class SkynetGPUConnector:
```

Comment:
We should discuss what name is apropiate, the way I would describe this class is that its a http client to three protocols:

- Telos (antelope protocol), both read and write (signing and pushing txs that alter chain state)
- Hyperion (extends antelope protocol read queries)
- IPFS, both to get inputs through HTTP gateway and publish results

Reply:
hmm so it's kinda like a (service) network `Client` but with different IPC proto specifics depending on service..

yeah i can't think of something offhand that's better but i do think there's prolly a (maybe novel semantic?)  word somewhere out there for this 😹

Reply:
Named it `NetConnector` now